app/uk/gov/hmrc/gform/core/parsers/ValueParser.scala

```diff
@@ -99,17 +115,26 @@ object ValueParser {
       contextField
     }
 
-  lazy val alphabeticOnly: Parser[String] = """\w+""".r ^^ { (loc, str) =>
+  lazy val alphabeticOnly: Parser[String] = """[a-zA-Z][a-zA-Z0-9]+""".r ^^ { (loc, str) =>
     str
   }
 
   lazy val anyConstant: Parser[Constant] = ("""[ \w,]+""".r ^^ { (loc, str) =>
     Constant(str)
   })
 
-  lazy val anyDigitConst: Parser[Expr] = (""" *[\d.][ \d.,]*""".r ^^ { (loc, str) =>
+  lazy val anyDigitConst: Parser[Expr] = ("""\d""".r ^^ { (loc, str) =>
```

Comment:
I have hard time to decode what we actually are parsing here.

These two regexp when removed don't even trigger any test failure:
```
"""\.[ \d]*\d""".r
"""\d[ \d,]*\.([ \d]*\d)?""".
```
It would be good next to test coverage to also have comment what we are parsing, couple of examples perhaps.

Reply:
This is needed in order to be able to parse "1." and ".1", but not accept just "."

Reply:
Have added more comments and tests